snuba/utils/streams/consumer.py

```diff
@@ -340,10 +326,11 @@ def poll(self, timeout: Optional[float] = None) -> Optional[KafkaMessage]:
             else:
                 raise ConsumerError(str(error))
 
-        result = KafkaMessage(
+        result = Message(
             Partition(Topic(message.topic()), message.partition()),
             message.offset(),
-            message.value(),
+            self.__codec.decode(Payload(message.key(), message.value())),
+            datetime.fromtimestamp(message.timestamp()[1] / 1000.0),  # XXX: tzinfo?
```

Comment:
I still need to figure out what to do here… it makes me nervous that all of our timestamps have no tzinfo. 😞

Reply:
is message.timsestamp() the timestamp the broker adds or the one we add when we produce ?

Reply:
It can be either, it depends on the configuration of `message.timestamp.type` on the broker: https://docs.confluent.io/current/clients/confluent-kafka-python/#confluent_kafka.Message.timestamp

Reply:
> I still need to figure out what to do here… it makes me nervous that all of our timestamps have no tzinfo. 😞

So, I don't think there is actually a correct answer for this question: specifically, whether or not to create this `datetime` object with `fromtimestamp` or `utcfromtimestamp`. All of the timestamps in Snuba are treated as naive, so they'll be treated as the local timestamp. Fortunately, the Docker image uses UTC as the local time zone, so the two functions are essentially equivalent. For local development (or anybody who runs this in something other than our provided Docker container in general), it's more complicated, since the local time zone will not necessarily be UTC.

Unfortunately, the method of instantiating `datetime` objects is a little inconsistent whether or not a `datetime` without a timezone actually represents the local time zone (as will happen via `datetime.now` or `datetime.fromtimestamp`) or UTC-adjusted time zone (`datetime.utcnow` and `datetime.utcfromtimestamp`) but more often than not, these timestamps seem to be instantiated explicitly as UTC, via:

1. `datetime.utcnow` (8 occurrences, compared to 0 of `datetime.now`),
2. `datetime.strptime` (7 occurrences), with the format as `%Y-%m-%dT%H:%M:%S.%fZ` which implies UTC due to the trailing `Z`,
3. `datetime.utcfromtimestamp` (2 occurrences, compared to 2 of `datetime.fromtimestamp`, of which only one is on a user-facing path)

`util.parse_datetime` (10 occurrences, mostly used in query processing) is a bit of a special case: https://github.com/getsentry/snuba/blob/355f63554811a2cd2f6f2dc659c46fbca3108189/snuba/util.py#L56-L58

It will just [outright discard all timezone information](https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.parse).

I'm going to update this so that the timestamp is instantiated with `utcfromtimestamp` for closest consistency with the existing implementations, but we should at some point take a step back and convert all of these into timezone aware datetimes — as far as I can tell, the only reason they weren't originally was because it was marginally harder to format the timezone in the format ClickHouse needed at the time that way.